pkg/anchor/writer/writer.go

```diff
@@ -545,59 +549,166 @@ func (c *Writer) getWitnesses(refs []*operation.Reference) ([]string, error) {
 			return nil, fmt.Errorf("unexpected interface '%T' for anchor origin", anchorOriginObj)
 		}
 
-		if strings.HasPrefix(anchorOrigin, "ipns://") {
-			resolvedAnchorOrigin, err := c.resolveAnchorOriginViaIPNS(anchorOrigin)
-			if err != nil {
-				return nil, fmt.Errorf("failed to resolve anchor origin via IPNS: %w", err)
-			}
-
-			anchorOrigin = resolvedAnchorOrigin
+		resolvedAnchorOrigin, err := c.resolveAnchorOrigin(anchorOrigin)
```

Comment:
Now that this PR evolved with "https" resolution as well I think that this functionality should be separate component e.g. resource resolver that will take in URL to resolve and property should be resolved against.

Reply:
Done!